File: cluster/midonet-cluster/src/main/java/org/midonet/cluster/rest_api/models/federation/VxlanSegment.java
/*
 * Copyright 2015 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.midonet.cluster.rest_api.models.federation;

import java.net.URI;
import java.util.List;
import java.util.UUID;
import java.util.concurrent.atomic.AtomicInteger;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;

import com.fasterxml.jackson.annotation.JsonIgnore;

import org.midonet.cluster.data.ZoomClass;
import org.midonet.cluster.data.ZoomField;
import org.midonet.cluster.models.Federation;
import org.midonet.cluster.rest_api.models.UriResource;
import org.midonet.cluster.util.UUIDUtil.Converter;

@ZoomClass(clazz = Federation.VxlanSegment.class)
public class VxlanSegment extends UriResource {

    @ZoomField(name = "id", converter = Converter.class)
    public UUID id;

    @ZoomField(name = "name")
    public String name;

    @Min(0)
    @Max(2 ^ 24 - 1)
    @ZoomField(name = "vni")
    public int vni;

    @JsonIgnore
    @ZoomField(name = "midonet_binding", converter = Converter.class)
    public List<UUID> midonetBindingIds;

    @JsonIgnore
    @ZoomField(name = "ovsdb_binding", converter = Converter.class)
    public List<UUID> ovsdbBindingIds;

    public URI getMidonetBindings() {
        return relativeUri(Application.MIDONET_BINDINGS);
    }

    public URI getOvsdbBindings() {
        return relativeUri(Application.OVSDB_BINDINGS);
    }

    @Override
    public URI getUri() {
        return absoluteUri(Application.VXLAN_SEGMENTS, id);
    }

    @JsonIgnore
    @Override
    public void create() {
        if (null == id) {
            id = UUID.randomUUID();
        }
    }

    @JsonIgnore
    public void update(VxlanSegment from) {
        this.id = from.id;
    }
}

File: python-midonetclient/src/midonetclient/federation/vxlan_segment.py
# vim: tabstop=4 shiftwidth=4 softtabstop=4

# Copyright 2015 Midokura SARL
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import logging

from midonetclient import resource_base
from midonetclient.federation import vendor_media_type

LOG = logging.getLogger(__name__)


class VxlanSegment(resource_base.ResourceBase):
    media_type = vendor_media_type.VXLAN_SEGMENT_JSON

    def __init__(self, uri, dto, auth):
        super(VxlanSegment, self).__init__(uri, dto, auth)

    def get_id(self):
        return self.dto['id']

    def get_name(self):
        return self.dto['name']

    def name(self, name):
        self.dto['name'] = name
        return self

    def get_group(self):
        return self.dto['group']

    def group(self, group):
        self.dto['group'] = group
        return self

    def get_vni(self):
        return self.dto['vni']

    def vni(self, vni):
        self.dto['vni'] = vni
        return self

    def get_subnet_prefix(self):
        return self.dto['subnetPrefix']

    def subnet_prefix(self, subnetPrefix):
        self.dto['subnetPrefix'] = subnetPrefix
        return self

    def get_subnet_length(self):
        return self.dto['subnetLength']

    def subnet_length(self, subnetLength):
        self.dto['subnetLength'] = subnetLength
        return self

        # self.put_attr(OvsdbBinding(name = 'ovsdb-bindings',
        #                         getter = 'get_ovsdb_bindings',
        #                         setter = 'ovsdb_bindings',
        #                         optional = True))
